src/content/docs/reference/stages.mdx

```diff
@@ -0,0 +1,55 @@
+---
+title: Stages
+description: A stage is a separate image build flow executed in parallel with the main build.
+---
+
+:::note
+This feature is currently only available when using `use_unstable_cli` option in the [GHA](/reference/github-action/)
+:::
+
+This property will allow users to define a list of Containerfile stages each with their own modules. Stages can be used to compile programs, perform parallel operations, and copy the results into the final image without contaminating the final image.
+
+## Module Support
+
+Currently the only modules that work out-of-the-box are `copy`, `script`, `files`, and `containerfile`. Other modules are dependent on the programs installed on the image. In order to better support some of our essential modules, a setup script is ran at the start of each stage that is not `scratch`. This script will install `curl`, `wget`, `bash`, and `grep` and use the package manager for the detected distributions.
+
+At this time, the following distributions are supported:
+
+- Debian
+- Ubuntu
+- Fedora
+- Alpine
+
+Contributions to increase the size of this list is [welcome](https://github.com/blue-build/cli)!
+
+## Syntax
+
+- **Required**
+  - `from` - The full image ref (image name + tag). This will be set in the `FROM` statement of the stage.
+  - `name` - The name of the stage. This is used when referencing the stage when using the `from:` property in the `copy` [module](/reference/modules/copy/).
+  - `modules` - The list of modules to execute. The exact same syntax used by the main recipe `modules:` property.
+- **Optional**
+  -  `shell` - Allows a user to pass in an array of strings that are passed directly into the [`SHELL` instruction](https://docs.docker.com/reference/dockerfile/#shell).
+
+### Example
+
+```yaml
+stages:
+- name: ubuntu-test
+  from: docker.io/library/ubuntu:22.04
+  modules:
+  - type: files
+    files:
+    - usr: /usr
+  - type: script
+    scripts:
+    - example.sh
+    snippets:
+    - echo "test" > /test.txt
+  - type: test-module
+  - type: containerfile
+    containerfiles:
+    - labels
+    snippets:
+    - RUN echo "This is a snippet"
+```
```

Comment:
Should be better documented. The indentation might also be wonky. Let me start the work for you:

```yaml
stages:
- name: ubuntu-stage
  from: docker.io/library/ubuntu:22.04
  modules:
  - type: files # why is usr copied here? usually it's copied into the final image and has stuff like systemd units and justfiles?
    files:
    - usr: /usr
  - type: script
    scripts:
    - build-from-source.sh # this script could for example git clone a repository, build it, and copy the result to /results/binaryname
    snippets: # useless example, but i don't know what i would replace it with
    - echo "test" > /test.txt
  - type: custom-module # runs a custom module
    output: /results/custom/
  - type: containerfile
    containerfiles:
    - labels # what use are labels in a stage, do they even get published anywhere?
    snippets:
    - RUN echo "This is a snippet" # this could be done with the script snippets, maybe there's something that can only be run with the containerfile module we could showcase here?

# i'd also add an example for how to copy things from the custom stage
modules:
- type: containerfile
  snippets:
  - COPY --from=ubuntu-stage /results/binaryname /usr/bin/binaryname
  - COPY --from=ubuntu-stage /results/custom/ /usr/share/custom-module-outputs/
```

Reply:
Yeah these were very much not good examples lol